docs/docs/theme-ui.md

```diff
@@ -0,0 +1,264 @@
+
+## Using Theme UI in Gatsby
+
+Theme UI includes the `gatsby-plugin-theme-ui` package to better integrate with your Gatsby project.
+
+Install the following packages to add Theme UI.
+
+```shell
+npm install theme-ui gatsby-plugin-theme-ui @emotion/core @mdx-js/react
+```
+
+After installing the dependencies, add the following to your `gatsby-config.js`.
+
+```js:title=gatsby-config.js
+module.exports = {
+  plugins: ["gatsby-plugin-theme-ui"],
+}
+```
+
+Theme UI uses a `theme` configuration object to provide color, typography, layout, and other shared stylistic values through [React context][].
+This allows components within your site to add styles based on a predefined set of values.
+
+The Theme UI plugin uses the [component shadowing API][] to add the theme object context to your site.
+Create a `src/gatsby-plugin-theme-ui` directory in your project, and add an `index.js` file to export a theme object.
+
+```shell
+mkdir src/gatsby-plugin-theme-ui
+```
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {}
+```
+
+## Creating a theme object
+
+Add a `colors` object to the file created above to store the color palette for your site.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+}
+```
+
+Next add some base typographic values.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+  // highlight-start
+  fonts: {
+    body: "system-ui, sans-serif",
+    heading: "system-ui, sans-serif",
+    monospace: "Menlo, monospace",
+  },
+  fontWeights: {
+    body: 400,
+    heading: 700,
+    bold: 700,
+  },
+  lineHeights: {
+    body: 1.5,
+    heading: 1.125,
+  },
+  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 72],
+  // highlight-end
+}
+```
+
+Next add values for use in margin and padding.
+
+```js:title=src/gatsby-plugin-theme-ui/index.js
+export default {
+  colors: {
+    text: "#333",
+    background: "#fff",
+    primary: "#639",
+    secondary: "#ff6347",
+  },
+  fonts: {
+    body: "system-ui, sans-serif",
+    heading: "system-ui, sans-serif",
+    monospace: "Menlo, monospace",
+  },
+  fontWeights: {
+    body: 400,
+    heading: 700,
+    bold: 700,
+  },
+  lineHeights: {
+    body: 1.5,
+    heading: 1.125,
+  },
+  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 72],
+  // highlight-start
+  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
+  // highlight-end
+}
+```
+
+Feel free to include as many additional values as you'd like in your theme.
+Read more about theming in the [Theme UI documentation](https://theme-ui.com/theming).
+
+## Adding styles to elements
+
+Theme UI uses a custom [JSX pragma][] comment to add support for Emotion's `css` prop,
```

Comment:
Since this term will be new to many people, is there a short way to summarize what it is? e.g. "a custom JSX pragma comment, a fancy type of function" or whatever the thing is? (i couldn't immediately tell from reading the Babel doc)

Reply:
Here's a suggestion based on [an explainer I read](https://jasonformat.com/wtf-is-jsx/): 

```suggestion
Theme UI uses a custom [JSX pragma][], a per-file or per-function comment to tell [Babel](/docs/glossary#babel) the name of Emotion's `css` prop.
```